### PR TITLE
Add pkg-config info for the riak-c-client

### PR DESCRIPTION
**Pull request by bookshelfdave**
Tested using a basic build:

```
./autogen.sh && ./configure && make && sudo make install
```

```
$ pkg-config --cflags --libs riak-c-client-sync
-lriak_c_client-0.5

$ pkg-config --cflags --libs riak-c-client-async
-D_THREAD_SAFE -I/usr/local/Cellar/libevent/2.0.21/include -L/usr/local/Cellar/libevent/2.0.21/lib -lriak_c_client-0.5 -levent_pthreads -levent
```

### CODE REVIEW

File: Makefile.scons [hunk 1]
```diff
@@ -1,5 +0,0 @@
-all:
```
Comment (hazen) [inline Makefile.scons:1]: Weird. I thought that file got nuked in #42

Comment (bookshelfdave) [inline Makefile.scons:1]: looks like it's still in develop https://github.com/basho/riak-c-client/blob/develop/Makefile.scons

**Comment by hazen** [comment]
So I tried installing this branch in `/tmp/riakc` just for fun and found that we are missing the include path for `riak-c-client` itself:

```
gcc `pkg-config --cflags --libs riak-c-client-sync` taste_of_riak.c -o taste_of_riak
taste_of_riak.c:21:10: fatal error: 'riak.h' file not found
#include <riak.h>
         ^
1 error generated.
make: *** [all] Error 1
```

It looks like the `-I` includes are not being returned.  I think the last line needs to be

```
Cflags: -I${includedir}
```

**Comment by hazen** [comment]
Also, how about simply `riak-c-client.pc` and `riak-c-client-async.pc`?  The default is synchronous, right?

**Comment by hazen** [comment]
Looks like we also need this:

```
Libs: -L${libdir} -lriak_c_client-0.5
```

**Comment by bookshelfdave** [comment]
@javajolt updated based on your feedback.

```
$ pkg-config --list-all | grep riak
riak-c-client                       Riak C Client, synchronous implementation - A C client for the Riak Distributed Database
riak-c-client-async                 Riak C Client, async implementation - A C client for the Riak Distributed Database
```

**Comment by bookshelfdave** [comment]
Note: I'll need to update the `Makefile` in taste-of-riak/C

**Comment by hazen** [comment]
Didn't we want to leave this open? I can't remember why.

Otherwise, it looks good :+1:

**Comment by bookshelfdave** [comment]
@javajolt I was going to leave it open while I got the Taste of Riak working. I'm thinking of implementing another simple async version to go alongside the sync one, so there might be more changes coming in the next few days.

**Pull request merged**